File: Level_2/올바른 괄호/solution.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int cnt = 0;
            
        for(int i = 0; i < s.length(); i++){
            
            if(s.charAt(i) == '(') cnt++;
            else cnt--;
            
            if(cnt < 0) return false;
        }
        
        if(cnt == 0) answer = true;
        else answer = false;

        return answer;
    }
}

File: Level_2/올바른 괄호/solution.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    int count = 0;
    
    for(int i = 0; i < s.length(); i++){
        string ch = s.substr(i, 1);
        
        if(ch == "(") count++;
        else count--;
        
        if(count < 0) return false;
    }
    
    if(count == 0) answer = true;
    else answer = false;
    

    return answer;
}
